src/webapi/date.rs

```diff
@@ -1,13 +1,117 @@
+        js!(
+            return new Date(@{year}, @{month}, @{day}, @{hours}, @{minutes}, @{seconds});
+        ).try_into().unwrap()
+    }
+    
+    /// Creates a JavaScript Date instance that represents a single moment in time.
+    /// Date objects are based on a time value that is the number of milliseconds since 1 January 1970 UTC.
+    /// 
+    /// year is an integer value representing the year. Values from 0 to 99 map to the years 1900 to 1999.
+    /// month is an integer value representing the month, beginning with 0 for January to 11 for December
+    /// day is an integer value representing the day of the month (normally from 1 to 31)
+    /// hours an integer value representing the minute segment of a time
+    /// seconds an integer value representing the second segment of a time
+    /// milliseconds an integer value representing the millisecond segment of a time
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date-year-month-date-hours-minutes-seconds-ms
+    pub fn new_with_time_and_millis(year: i32, month: i32, day: i32, hours: i32, minutes: i32, seconds: i32, milliseconds: i32) -> Self {
+        js!(
+            return new Date(@{year}, @{month}, @{day}, @{hours}, @{minutes}, @{seconds}, @{milliseconds});
+        ).try_into().unwrap()
+    }
+
+    /// Creates a JavaScript Date instance that represents a single moment in time.
+    /// Date objects are based on a time value that is the number of milliseconds since 1 January 1970 UTC.
+    /// 
+    /// String value representing a date. The string should be in a format recognized by
+    /// the Date.parse() method (IETF-compliant RFC 2822 timestamps and also a version of ISO8601).
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date-value
+    pub fn new_with_string(date_string: &str) -> Self {
+        js!(
+            return new Date(@{date_string});
+        ).try_into().unwrap()
+    }
+    
+    /// Creates a JavaScript Date instance that represents a single moment in time.
+    /// Date objects are based on a time value that is the number of milliseconds since 1 January 1970 UTC.
+    /// 
+    /// Integer value representing the number of milliseconds since January 1, 1970, 00:00:00 UTC,
+    /// with leap seconds ignored (Unix Epoch; but consider that most Unix timestamp functions count in seconds).
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
+    // https://www.ecma-international.org/ecma-262/6.0/#sec-date-value
+    pub fn new_with_now(now: f64) -> Self {
+        js!(
+            return new Date(@{now});
+        ).try_into().unwrap()
+    }
```

Comment:
Hmm.... I'd like a better name for this one, but it's somewhat hard to come up with one. `from_time`? (To be consistent with `set_time`.) The `new_with_now` sounds kinda strange.